packages/shared/routes/dashboard/governance/views/GovernanceEventDetails.svelte

```diff
@@ -96,6 +108,54 @@
         )
         currentVoteValue = participation?.answers[0] ?? null
     }
+
+    const handleTransferState = (state: TransferState): void => {
+        if (!state) return
+
+        const _onCancel = () => {
+            transferState.set(null)
+
+            closePopup(true)
+        }
+
+        const { data, type } = state
+
+        switch (type) {
+            case TransferProgressEventType.PerformingPoW:
+                closePopup(true)
+                break
+            case TransferProgressEventType.SigningTransaction:
+                ledgerAwaitingConfirmation = true
+                openPopup(
+                    {
+                        type: 'ledgerTransaction',
+                        hideClose: true,
+                        preventClose: true,
+                        props: {
+                            ...handleTransactionEventData(transactionEventData),
+                            onCancel: _onCancel,
+                        },
+                    },
+                    true
+                )
+                break
+            case TransferProgressEventType.PreparedTransaction:
+                transactionEventData = data
+                break
+            default:
+                break
+        }
+    }
+
+    const unsubscribeFromTransferState = transferState.subscribe((state) => {
```

Comment:
We could shorten the following code as follows to leverage auto-subscriptions from Svelte and remove the onDestroy hook.

```
$: $transferState, handleLedgerTransfer()
const handleLedgerTransferState = () => {
  if ...
}
```